packages/jest-matchers/src/spy-matchers.js

```diff
@@ -44,6 +45,24 @@ const spyMatchers: MatchersObject = {
 
     return {message, pass};
   },
+
+  toHaveBeenCalledWith(actual: any, expected: any) {
+    ensureSpy(actual, 'toHaveBeenCalledWith');
+
+    const args = Array.from(arguments).pop().args;
+    const pass = actual.calls.all().some(call => {
+      return Object.keys(args).every((key, i) => {
+        return args[key] === call.args[i];
+      });
+    });
+    const message = pass
+      ? `expected the spy to not be called` +
+        ` with '${stringify(args)}' but it was`
+      : `expected the spy to be called with` +
+        ` '${stringify(args)}', but it wasn't`;
```

Comment:
you should add a list of calls if there are any..
print 3 recent calls and `x more` if there's more than 3